src/mlir/dialect/VeronaTypes.h

```diff
@@ -16,6 +16,18 @@ namespace mlir::verona
   /// dialect.
   bool areVeronaTypes(llvm::ArrayRef<Type> types);
 
+  /// Normalize a type by distributing unions and intersections, putting the
+  /// type in disjunctive normal form. This is a necessary step in order for
+  /// subtyping to recognise certain relations.
+  ///
+  /// The amount of normalization done is quite limited. In particular it does
```

Comment:
I would move this type of comment inside the implementation, as a TODO maybe.

Reply:
I consider what this function does or does not normalize part of its contract, and not an implementation detail, so I think it is relevant to have it in the header file.

Reply:
The first paragraph, yes. The second, sounds like a TODO to me.

If not, then we won't do any flattening, and we'll have the undesirable property of: `join<A, join<B, C>> != join<join<A, B>, C> != join<A, B, C>`. Same for simplification, we'll end up having to do that kind of match elsewhere.

I'd rather normalise them all in a single place.

Reply:
I didn't mean we shouldn't change that contract ever, only that this is merely its current contract. But I guess it's a minor detail of it, so I've moved it out of the header.